packages/diagrams/frontend/sirius-components-diagrams/src/renderer/layout/useSynchronizeLayoutData.ts

```diff
@@ -101,9 +104,57 @@ export const useSynchronizeLayoutData = (): UseSynchronizeLayoutDataValue => {
           resizedByUser,
         });
       }
+      const outsideLabelPosition = Object.values(node.data.outsideLabels)[0];
+      if (outsideLabelPosition) {
+        labelLayoutData.push({
+          id: outsideLabelPosition.id,
+          position: {
+            x: outsideLabelPosition.position.x,
+            y: outsideLabelPosition.position.y,
+          },
+        });
+      }
+    });
+
+    diagram.edges.forEach((edge) => {
+      const centerLabel = edge.data?.label;
+      if (centerLabel) {
+        labelLayoutData.push({
+          id: centerLabel.id,
+          position: {
+            x: centerLabel.position.x,
+            y: centerLabel.position.y,
+          },
+        });
+      }
+      if (edge.data && 'beginLabel' in edge.data) {
+        const beginLabel = edge.data?.beginLabel as EdgeLabel;
```

Comment:
You are testing twice that edge.data is present (same comment line 142), you could remove an "if" each time